utils/resource/usage.go

```diff
@@ -150,6 +155,19 @@ func (m *manager) TrackProcess(pid int) {
 	m.processesLock.Unlock()
 }
 
+func (m *manager) HasProcess(pid int) bool {
+	m.processesLock.Lock()
+	defer m.processesLock.Unlock()
+	proc, ok := m.processes[pid]
+	if !ok {
+		return false
+	}
+	p, _ := os.FindProcess(int(proc.p.Pid))
+	// To properly check if a process exists, we need to send signal 0.
+	// https://pkg.go.dev/os#FindProcess
+	return p.Signal(syscall.Signal(0)) == nil
```

Comment:
Is `p` guaranteed to be non-nil on non-unix systems?

Reply:
good catch https://github.com/golang/go/blob/master/src/os/exec_windows.go#L84